### ISSUE TITLE
should callouts handle multiple angles?

### ISSUE THREAD

**Issue by lindsayplatt** [USGS-R/gsplot#301]
Also, should we handle if some of those are auto vs defined? We would need to convert non-autos into numeric inside the callouts function

```
gs <- gsplot(xaxs='r') %>% 
    points(1:5, 1:5) %>% 
    callouts(c(4, 3), c(2, 3), labels="stuff", angle = c('auto', '150'))
gs

Warning message:
In if (angle != "auto") { :
  the condition has length > 1 and only the first element will be used
```

**Comment by lindsayplatt** [comment]
@jread-usgs @jiwalker-usgs @ldecicco-USGS what do you think? I want to implement if the answer is yes because repgen is waiting on this. If no, I will alter the repgen code

**Comment by jordansread** [comment]
If I am understanding this correctly, yes, I see no reason that `angle` can't be a vector of the same length as `x`

**Comment by lindsayplatt** [comment]
Ok, then my followup question is should users be able to specify "auto". So if they knew two of the angles but didn't want to specify a third -- c(45, 150, 'auto'). I know this would coerce the other two numbers into character, so I don't think this should be the solution. But a case that I wanted to bring up

**Comment by jordansread** [comment]
is that a driving use-case for the change now? or could we fail when it is a char vector of length > 1, and allow it when the vector is numeric?

**Comment by lindsayplatt** [comment]
No, not necessary now - I'll use your suggestion for now

**Comment by ldecicco-USGS** [comment]
Without thinking about it too much, I'd vote the user would specify c(45, 150, NA) or just "auto" or NA (if the arg is either "auto" or all NA's...that's the equivalent of auto, otherwise it's a numeric vector).

**Comment by jiwalker-usgs** [comment]
I like the idea that auto = rep(NA, length(x))

**Comment by lindsayplatt** [comment]
@jiwalker-usgs are you suggesting that as an additional argument in callouts? Or internally, we accept anything with NA as "auto"?

This is how I envision @ldecicco-USGS's suggestion, and I like it. But is having "NA" == "auto" intuitive to most people?

```
callouts(c(4, 3), c(2, 3), labels="stuff", angle = c(NA, 150))
```

**Comment by jiwalker-usgs** [comment]
No additional arguments, just that angle takes either "auto" or a numeric vector where NA means auto.  I think it is intuitive enough.

**Comment by jordansread** [comment]
hmmm...I thought `NA` meant to skip the callout for that x val

**Comment by jiwalker-usgs** [comment]
ooh, didn't think of that

**Comment by lindsayplatt** [comment]
Isn't that why `labels = NA`?

If callouts aren't used for a particular x/y because `angle = NA`, I think we need a separate argument for specifying auto. 

If `auto == TRUE` and `angle == NA`, it will auto find the angle. 
If `auto == FALSE` and `angle == NA`, it doesn't plot that callout. 
If `auto == FALSE` and `angle == 45`, it just uses that angle. 

The only question is what if someone put `auto == TRUE` and `angle == 45` -- what takes priority?

**Comment by jordansread** [comment]
Easy solution is you are either using auto for all, or you are specifying where you want them to be. Do we need to make auto smarter?

**Comment by jiwalker-usgs** [comment]
I'm going back and forth on this now.  I think angle = NA means auto angle.  I think @lindsaycarr is right, labels should be a vector that can enable/disable callouts on particular points.  Angle on these would then be ignored. Maybe?

**Comment by jordansread** [comment]
ok, but you can have the callout line w/o a label, which would be `NA`, right?

**Comment by lindsayplatt** [comment]
hmm good point. Would it be reasonable to expect that if they don't want a callout line or label for a specific point, then they shouldn't include that point in the callouts function?

**Comment by jiwalker-usgs** [comment]
The problem with that is callouts can be used within a points call where the points are passed right through to it.  Hmm.

**Comment by lindsayplatt** [comment]
oh yeah, forgot about embedded functions

**Comment by lindsayplatt** [comment]
@jiwalker-usgs and I just talked and here's what we think:
- `labels = NA` should be used to specify if you don't want to have a callout plotted for a particular x/y. 
- `angle = NA` or `angle = 'auto'` specifies that the function will determine the best angle for the callout line

and if people really want a callout line with no label (which seems like a pretty small use case), they could use an empty string: `labels = ""`

```
#example 1
gs <- gsplot() %>% 
    callouts(1:2,1:2, labels = c(NA, "stuff"), angle = c(NA, 150))
gs

#example 2
gs <- gsplot() %>% 
    callouts(1:2,1:2, labels = c("", "stuff"), angle = c(NA, 150))
gs
```
- Example 1 above would show a callout on (2,2) at an angle of 150 that says "stuff". 
- Example 2 above would show a callout on (2,2) at an angle of 150 that says "stuff" & a callout on (1,1) with no label at an angle determined internally.

**Comment by jordansread** [comment]
:+1:

**Comment by lindsayplatt** [comment]
fix for this was added in #305
